src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,201 @@
+{
+    protected $title = 'Members with MFA enabled on their account';
+
+    protected $description = 'Identifies all members on this site, with columns added to indicate the status each has'
+        . ' in regards to multi factor authentication method registration. E.g. how many methods are set up by that'
+        . ' member, which method types, and who has opted not to set up MFA (if this option is enabled for the site)';
+
+    protected $dataClass = Member::class;
+
+    public function title()
+    {
+        return _t(__CLASS__ . '.TITLE', parent::title());
+    }
+
+    public function description()
+    {
+        return _t(__CLASS__ . '.DESCRIPTION', parent::description());
+    }
+
+    /**
+     * @param array $params
+     */
+    public function sourceRecords($params)
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                // When querying SQL COUNT() - this will always factor in the backup method, which is excluded in the
+                // secondary query below to get the list of method names the user has registered. So to compensate we
+                // need to bump up the entered number by 1.
+                $params['Count'] = $desiredCount + 1;
+            }
+        }
+
+        $filteredList = ArrayList::create([]);
+        $backupClass = get_class(singleton(MethodRegistry::class)->getBackupMethod());
+        /** @var Member[]&MemberExtension[] $members */
+        $members = $this->applyParams(Member::get(), $params)->toArray();
+        foreach ($members as $member) {
+            $defaultMethod = $member->getDefaultRegisteredMethod();
+            $defaultMethodClassName = $defaultMethod ? $defaultMethod->MethodClassName : '';
+
+            $registeredMethods = $member
+                ->RegisteredMFAMethods()
+                ->exclude('MethodClassName', $backupClass)
+                ->column('MethodClassName');
+            $registeredMethodNames = array_map(function (string $methodClass) use ($methods) {
+                return $methods[$methodClass];
+            }, $registeredMethods);
+
+            $memberReportData = ArrayData::create();
+            foreach ($member->summaryFields() as $field => $name) {
+                $memberReportData->$field = $member->$field;
+            }
+
+            $memberReportData->methodCount = (string)(count($registeredMethods) ?: '');
```

Comment:
Re: previous conversation about wanting a number or an empty string, I'm not sure I follow why - surely showing "Registered methods: 0" is not a bad thing, rather than not showing a number at all?

Reply:
Personally I find it visually disruptive to scan a column for a valid entry amongst a list of invalid ones, rather than having them stand out easily. Granted there is filtering, but I cannot range with this filter. Why not provide a better interface where we can?

Reply:
"Better" is subjective. IMO if you're omitting data from a column it's not better. For example, if I wanted to use this report in a CSV and pass it to some other system, I'd have to write logic to handle an empty value in a specific column as zero. In other cases it might be "empty value = yes" instead, etc. I don't care enough about it to keep this point open though.